src/commands/synthetics/reporters/junit.ts

```diff
@@ -455,23 +487,24 @@ export class JUnitReporter implements Reporter {
     const test = result.test
     const resultOutcome = getResultOutcome(result)
     const resultUrl = getResultUrl(baseUrl, test, result.resultId)
-    const passed = [ResultOutcome.Passed, ResultOutcome.PassedNonBlocking].includes(resultOutcome)
+
+    const passed = PASSED_RESULT_OUTCOMES.includes(resultOutcome)
 
     const id = `id: ${test.public_id}`
-    const location = `location: ${result.location}`
-    const device = isDeviceIdSet(result.result) ? ` - device: ${result.result.device.id}` : ''
+    const location = hasResult(result) ? `location: ${result.location}` : ''
+    const device = hasResult(result) && isDeviceIdSet(result.result) ? ` - device: ${result.result.device.id}` : ''
     const resultTimedOut = result.timedOut ? ` - result id: ${result.resultId} (not yet received)` : ''
 
     // This has to identify results, otherwise GitLab will only show the last result with the same name.
-    const resultIdentification = `${test.name} - ${id} - ${location}${device}${resultTimedOut}`
+    const resultIdentification = getResultIdentification(test, id, location, device, resultTimedOut)
 
     return {
       $: {
         classname: test.suite,
         file: test.suite,
         name: resultIdentification,
-        time: getResultDuration(result.result) / 1000,
-        timestamp: new Date(result.timestamp).toISOString(),
+        time: hasResult(result) ? getResultDuration(result.result) / 1000 : 0,
+        timestamp: hasResult(result) ? new Date(result.timestamp).toISOString() : new Date().toISOString(),
```

Comment:
I didn't invent `0` and `new Date().toISOString()`. They come from `getSkippedTestCase()` above in the file:

https://github.com/DataDog/datadog-ci/blob/1d8e7821d132076a3237dc86f3bb185a355f3814/src/commands/synthetics/reporters/junit.ts#L442-L443